project-structure/project-interface.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Project interface
+description: The standard interface for our projects - package.json and entrypoint
+---
+
+Our projects should all include [a `package.json` file](https://yarnpkg.com/lang/en/docs/package-json/) at the top-level of the project, which defines the standard ways the project should be run.
+
+The `package.json` file should define some [`scripts`](https://yarnpkg.com/lang/en/docs/package-json/#toc-scripts), and (usually) some [`dependencies` and `devDependencies`](https://yarnpkg.com/lang/en/docs/package-json/#toc-dependencies).
+
+## scripts
+
+These defined `scripts` can be run either with `yarn run {script-name}` or through [our local development tooling](https://canonical-web-and-design.github.io/practices/project-structure/the-run-script.html).
+
+Our standard script names:
+
+- `start`: The standard entrypoint for local development. It should run any required watchers for rebuilding files as they change, and run any local development servers etc..
+    - E.g.: `concurrently 'yarn run watch' 'yarn run serve'`
+- `build`: Build all the required assets for the project to be published. This may simply run a number of other scripts per filetype, e.g.:
+    - `build-js`: Build the JavaScript files
+    - `build-css`: Build any SCSS files etc. into CSS
+- `serve`: Run the a local webserver. This will typically use the same `./entrypoint` script as is used by the production Docker image (see below).
+- `watch`: Watch for changes in any relevant files. Again, this may be split up into sup-scripts by filetype, e.g.:
+    - `watch-js`: Watch for changes in JavaScript files and rebuild as needed.
+    - `watch-css`: Watch for changes in SCSS files and rebuild CSS as needed.
```

Comment:
- `test`
  - `test-py`
  - `test-js`
- `lint`
  - `lint-py`
  - `lint-js`
  - `lint-css`
?

Reply:
I writed some things about this.